gpu/octree/src/cuda/radius_search.cu

```diff
@@ -277,14 +293,15 @@ namespace pcl
                 for(;;)
                 {                
                     int take = 0;
+                    float d2 = 0.f;
```

Comment:
This should be initialized to nan, to prevent any misinterpretation of what this distance represents.